test/jdk/java/lang/Thread/virtual/InterruptHttp.java

```diff
@@ -55,7 +55,9 @@ public static void main(String[] args) throws Exception {
             });
 
             // give time for thread to block waiting for HTTP server
-            Thread.sleep(1000);
+            while (thread.getState() != Thread.State.WAITING) {
+                Thread.sleep(1000);
+            }
```

Comment:
Thanks for this. You are right that that the test is problematic. Can you try the following instead of polling the thread state? This will wait until a connection is established so that the read is interrupted.

```
while (server.connectionCount() == 0) {
    Thread.sleep(100);
}
```

Reply:
Thanks for your review.
Do I need remove this? I think the following is duplicated with the check of "server.connectionCount() == 0"
```
    // server should have accepted one connection
    if (server.connectionCount() != 1)
        throw new RuntimeException("Expected 1 HTTP connection");
```

Reply:
We should keep it to checks that only one connection is established. The details on this are in sun.net.www.http.HttpClient where it makes a decision on whether to retry the HTTP request when interrupted. In this case, it should not retry.

Reply:
Got it. Thanks for your answer.
I have changed this PR on your suggestion.